src/view/MapVisualizationView.java

```diff
@@ -1,2 +1,240 @@
+        panelTop.add(fitButton);
+
+        // checkbox to show satellite imagery
+        final JCheckBox showImageryCheckBox = new JCheckBox("Show satellite imagery");
+        showImageryCheckBox.addActionListener(new ActionListener() {
+            @Override
+            public void actionPerformed(ActionEvent e) {
+                if (showImageryCheckBox.isSelected())
+                    map.setTileSource((new BingAerialTileSource()));
+                else if (!(showImageryCheckBox.isSelected()))
+                    map.setTileSource((new OsmTileSource.Mapnik()));
+            }
+        });
+        panelTop.add(showImageryCheckBox);
+
+        // add the actual map
+        add(map, BorderLayout.CENTER);
+
+        // detects mouse click events
+        map.addMouseListener(new MouseAdapter() {
+            @Override
+            public void mouseClicked(MouseEvent e) {
+                if (e.getButton() == MouseEvent.BUTTON1) {
+                    map.getAttribution().handleAttribution(e.getPoint(), true);
+                }
+            }
+        });
+
+        // detects mouse movement to shift map accordingly
+        map.addMouseMotionListener(new MouseAdapter() {
+            @Override
+            public void mouseMoved(MouseEvent e) {
+                Point p = e.getPoint();
+                boolean cursorHand = map.getAttribution().handleAttributionCursor(p);
+                if (cursorHand) {
+                    map.setCursor(new Cursor(Cursor.HAND_CURSOR));
+                } else {
+                    map.setCursor(new Cursor(Cursor.DEFAULT_CURSOR));
+                }
+            }
+        });
+
+    // layer management (show buses and trains)
+    LayerGroup vehicles = new LayerGroup("Vehicles");
+    Layer trainsLayer = vehicles.addLayer("Trains");
+    VisualizeState vehicleData = visualizeViewModel.getVisualizationState();
+
+    JComboBox<Coordinate> vehicleSelector =
+            new JComboBox<>(vehicleData.getCoordinateList().toArray(new Coordinate[vehicleData.getSize()]));
+    vehicleSelector.setRenderer(new myRenderer());      // add custom renderer
+    vehicleSelector.addItemListener(new ItemListener() {
+        @Override
+        public void itemStateChanged(ItemEvent e) {
+            Coordinate coordinate = (Coordinate) e.getItem();
+            map.setDisplayPosition(coordinate, map.getZoom());
+        }
+    });
+    panelTop.add(vehicleSelector);
+
+    // add markers
+    for (int i = 0; i < vehicleData.getSize(); i++) {
+        MapMarkerDot marker = new MapMarkerDot(
+                trainsLayer,
+                vehicleData.getStringList().get(i),
+                vehicleData.getCoordinateList().get(i),
+                this.defaultStyle
+        );
+        map.addMapMarker(marker);
+    }
+
+    // On initialization, map is focused on the first vehicle within the list
+    // TODO: Implement properly
+    // TODO: What happens if there are no vehicles in the list? Throw an error?
```

Comment:
If I recall correctly, in @jacqnnn's use case, if there are NO incoming vehicles, an error dialog button would pop up when the user clicks "Show Incoming Vehicles". Therefore, her panel would never display should this happen. 

Since your use case is only triggered when her panel is shown AND when the map button is pressed, I think this case would be already handled by @jacqnnn .

In terms of unit test, you could write a fail test where if there are no vehicles in the list, then in `VisualizeOutputData`, attributes like `coordinateData` are empty for example